### ISSUE TITLE
invalid Requires in cockpit.spec

### ISSUE THREAD

**Issue by smwltr** [cockpit-project/cockpit#15891]
Cockpit version: master branch
OS: Fedora

On line 651 of tools/cockpit.spec there is:
Requires: %{__python3}
This does not get replaced by tools/gen-spec-dependencies.
I've tried it with:
Requires: python3
and it works fine on Fedora 34.

Steps to reproduce

1. # tools/gen-spec-dependencies tools/cockpit.spec
2. # dnf builddep --spec tools/cockpit.spec

Result:
RPM: error: line 651: Dependency tokens must begin with alpha-numeric, '_' or '/': Requires: %{__python3}
Failed to open: 'tools/cockpit.spec', not a valid spec file: can't parse specfile

If tools/gen-spec-dependencies is suppose to replace this line, then tools/gen-spec-dependencies needs to have code for that.

PR #15890 fixes this.

**Comment by marusak** [comment]
You are missing `python-srpm-macros` package. Can you please try to install it and see if it fixes your problem?

If it fixes your issue, then the question is, what were you trying to do. I __think__ that all current developers including CI have this installed due to it being pulled in as dependency of `rpm-build`. We already document that new developers should install this package - but not in this context. So if you believe that what you were trying to do should be documented, can you please push it as another commit  in https://github.com/cockpit-project/cockpit/pull/15888 (as it should be repushed with "fixes" line anyway).

**Comment by smwltr** [comment]
Yes, that package was not installed. I'll add it to the HACKING.md and make another PR.

Another related thing: I think the sed 's/%{npm-version:.*}/0/' > temp file etc. commands may not be needed anymore. Everything ran fine with just: 
sudo dnf builddep --spec tools/cockpit.spec
As it looks like tools/gen-spec-dependencies replaces npm-version with the actual version number.
I may be totally mistaken about any of this.

**Comment by marusak** [comment]
> and make another PR

Please just force push into  #15890 or #15888 

> I think the sed 's/%{npm-version:.*}/0/' > temp file etc. commands may not be needed anymore. Everything ran fine with just:

Right, the example in HACKING.md is not using `tools/gen-spec-dependencies`. If you can rewrite it to use this script instead of all of those temp files and `sed`s that would be great :)

**Comment by smwltr** [comment]
OK, I will force pushed to #15888, as all changes are in HACKING.md, but it shows 2 commits. Maybe because I used --force-with-lease or maybe I should have rebased on my WC. Sorry about that.
